### PR TITLE
Offline signer + demo certificate

### PR DESCRIPTION
**Pull request by jirihelmich**

### CODE REVIEW

File: src/Mews.Registrierkassen.Tests/Mews.Registrierkassen.Tests.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>Mews.Registrierkassen.Tests</RootNamespace>
     <AssemblyName>Mews.Registrierkassen.Tests</AssemblyName>
-    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
```
Comment (ondrahermanek) [inline src/Mews.Registrierkassen.Tests/Mews.Registrierkassen.Tests.csproj]: 4.7?

File: src/Mews.Registrierkassen/OfflineSigner.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+namespace Mews.Registrierkassen
+{
+    public class OfflineSigner
+    {
+        public OfflineSigner(X509Certificate2 certificate)
+        {
+            Certificate = certificate;
+        }
+
+        public X509Certificate2 Certificate { get; set; }
```
Comment (ondrahermanek) [inline src/Mews.Registrierkassen/OfflineSigner.cs]: ~~set;~~ to keep the class immutable.

File: src/Mews.Registrierkassen/OfflineSigner.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            //// This is a manual JWS implementation as RKSV does not use standard signature format. Do not migrate to jose-jwt
+            var jwsHeaderBase64Url = "eyJhbGciOiJFUzI1NiJ9"; // Fixed value for RKSV
+            var jwsPayloadBase64Url = Base64UrlEncoder.Encode(qrData.Value);
+            var jwsDataToBeSigned = jwsHeaderBase64Url + "." + jwsPayloadBase64Url;
+
+            var bytes = Certificate.GetECDsaPrivateKey().SignData(Encoding.UTF8.GetBytes(jwsDataToBeSigned), HashAlgorithmName.SHA256);
+            var jwsSignatureBase64Url = Base64UrlEncoder.Encode(bytes);
+
+            var signerOutput = jwsHeaderBase64Url + "." + jwsPayloadBase64Url + "." + jwsSignatureBase64Url;
```
Comment (vitasamek) [inline src/Mews.Registrierkassen/OfflineSigner.cs]: `$"{jwsHeaderBase64Url}.{jwsPayloadBase64Url}.{jwsSignatureBase64Url}"`?

File: src/Mews.Registrierkassen/OfflineSigner.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            Certificate = certificate;
+        }
+
+        public X509Certificate2 Certificate { get; }
+
+        public SignerOutput Sign(QrData qrData)
+        {
+            //// This is a manual JWS implementation as RKSV does not use standard signature format. Do not migrate to jose-jwt
+            var jwsHeaderBase64Url = "eyJhbGciOiJFUzI1NiJ9"; // Fixed value for RKSV
```
Comment (siroky) [inline src/Mews.Registrierkassen/OfflineSigner.cs]: Base64UrlEncoder.Encode("{"alg":"ES256"}");

File: src/Mews.Registrierkassen/OfflineSigner.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            var jwsPayloadBase64Url = Base64UrlEncoder.Encode(qrData.Value);
```
Comment (siroky) [inline src/Mews.Registrierkassen/OfflineSigner.cs]: Out of curiosity, how does this differ from plain base-64? If it doesn't then drop the Url suffix.

File: src/Mews.Registrierkassen.Tests/OfflineTests.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                    standartRate: new CurrencyValue(29.73m),
+                    lowerReducedRate: new CurrencyValue(36.41m),
+                    specialRate: new CurrencyValue(21.19m)
+                ),
+                turnover: new CurrencyValue(0.0m),
+                certificateSerialNumber: new CertificateSerialNumber(certificate.SerialNumber),
+                previousJwsRepresentation: new JwsRepresentation("d3YUbS4CoRo="),
+                key: Convert.FromBase64String("RCsRmHn5tkLQrRpiZq2ucwPpwvHJLiMgLvwrwEImddI="),
+                created: new LocalDateTime(
+                    new DateTime(2015, 11, 25, 19, 20, 11),
```
Comment (vitasamek) [inline src/Mews.Registrierkassen.Tests/OfflineTests.cs]: name values?

File: src/Mews.Registrierkassen/OfflineSigner.cs (hunk 3 above)
Comment (siroky) [inline src/Mews.Registrierkassen/OfflineSigner.cs]: String interpolation here and on line 23.

**Pull request merged**